packages/flutter/lib/src/painting/image_provider.dart

```diff
@@ -1407,10 +1407,42 @@ class ResizeImage extends ImageProvider<ResizeImageKey> {
   }
 }
 
+/// The strategy for [Image.network] and [NetworkImage] to decide whether to
+/// display images in <img> elements in a platform view instead of fetching
+/// bytes.
+///
+/// See [Image.network] for more explanation on the impact.
+///
+/// This option is only effective on the Web platform. Other platforms always
+/// display network images by fetching bytes.
+enum WebImgElementStrategy {
+  /// Never use <img> elements.
+  never,
+
+  /// Use <img> elements when fetching bytes is not available.
+  ///
+  /// This strategy uses <img> elements if `header` is empty and the fetch
+  /// encounters errors.
+  whenNecessary,
+
+  /// Always use <img> elements as long as `header` is empty.
+  ///
+  /// This strategy still fetches bytes if `header` is not empty, since <img>
```

Comment:
typo:
```suggestion
  /// Always use <img> elements as long as `headers` is empty.
  ///
  /// This strategy still fetches bytes if `headers` is not empty, since <img>
```